### ISSUE TITLE
false single quotes instead of double quotes; even formatter

### ISSUE THREAD

**Issue by Reissner** [nvuillam/npm-groovy-lint#229]
I have the following code snippet: 
```
  for ( int i = 1; i <= NrOfTries; i++ ) {
    try {
      echo "Trying to start Matlab. Try nr: ${i}"
      timeout(time: 1, unit: 'MINUTES') {
        bat """
            "${env.MatlabPath}matlab" -wait -batch ${matlabCall}
        """
      }
```
There i get a message that in `bat` one can transform triple double quotes into single quotes which is wrong. 
What makes things worse: The formatter perfroms the transformation really.

**Comment by nvuillam** [comment]
The problem is that CodeNarc triggers a false positive , and formatter catches it and applies the update
i'm curently upgrading to CodeNarc v3.1.0, if the problem is still here (it's possible because I don't see updates about UnnecessaryGString in 3.1.0 release notes), please can you report the issue there ? https://github.com/CodeNarc/CodeNarc

**Comment by nvuillam** [comment]
This seems ok with new version

@Reissner please can you confirm ? :) thanks !

Before:
![image](https://user-images.githubusercontent.com/17500430/184503722-144941c4-8106-4997-9b1d-5c776094aa48.png)

After:
![image](https://user-images.githubusercontent.com/17500430/184503729-444eb339-2098-4761-b9c8-0c0325d17e8b.png)

**Comment by edco29** [comment]
hi @nvuillam  I getting the same issue , I'm using the groovy lint from visual studio code and when i save my changes , the plugin automatically convert double quotes to single even in scenarios describes above where is necessary to have a """ ( triple double quotes)  ( btw I'm using jenkisnfile)

**Comment by nvuillam** [comment]
Unfortunately I don't have a workaround :(
